### PR TITLE
Add bartender skill chip and beer goggles blur effect

### PR DESCRIPTION
**Pull request by timothymtorres**
## About The Pull Request

- Adds new bartender skill chip that lets them scan reagents and throw drinks without spilling
- Removes the ability to scan reagents and throw drinks without spilling using beer goggles
- Adds drunk blur effect while wearing beer goggles
- Changes bartender glasses to be plain sunglasses instead of beer goggles

## Why It's Good For The Game

Just like how the chef has CQC, the bartender will now spawn with the ability to see reagents in drinks and toss drinks around.  This used to be tied to the beer goggles itself which didn't make any sense.  Now beer goggles have the blur effect that is more of a gag item for people to use for gimmicks.

## Changelog

:cl:
add: Adds new bartender skill chip that lets them scan reagents and throw drinks without spilling
add: Adds drunk blur effect while wearing beer goggles
add: Changes bartender glasses to be plain sunglasses instead of beer goggles
del: Removes the ability to scan reagents and throw drinks without spilling using beer goggles
/:cl:

### CODE REVIEW

**Comment by private-tristan** [comment]
so now bartenders now need to choose between
> seeing funny the entire shift
> giving up flash protection
> rushing tools to make regular sunglasses out of beergoggles

maybe make the blur effect tied to alt clicking the bartender glasses (sorta like the color thingy that i've never seen anyone use)

**Comment by Colovorat** [comment]
> so now bartenders now need to choose between
> 
> > seeing funny the entire shift
> > giving up flash protection
> > rushing tools to make regular sunglasses out of beergoggles
> 
> maybe make the blur effect tied to alt clicking the bartender glasses (sorta like the color thingy that i've never seen anyone use)

Literally says that bartender now spawns with regular sunglasses, what's your problem

File: code/modules/clothing/glasses/_glasses.dm [hunk 1]
```diff
@@ -307,7 +307,15 @@
 	name = "beer goggles"
 	icon_state = "sunhudbeer"
 	desc = "A pair of sunglasses outfitted with apparatus to scan reagents, as well as providing an innate understanding of liquid viscosity while in motion."
-	clothing_traits = list(TRAIT_BOOZE_SLIDER, TRAIT_REAGENT_SCANNER)
+	clothing_traits = list(TRAIT_DRUNK_VISION)
+
+/obj/item/clothing/glasses/sunglasses/reagent/equipped(mob/living/carbon/human/user, slot)
+	. = ..()
+	user.update_eye_blur()
```
Comment (MrMelbert) [inline code/modules/clothing/glasses/_glasses.dm:314]: You need to check if it was equipped in the right slot in `equipped`, or else it'll apply when in hand

**Comment by private-tristan** [comment]
> Literally says that bartender now spawns with regular sunglasses, what's your problem

oopsies I am big dumb

File: code/modules/clothing/glasses/_glasses.dm [hunk 2]
```diff
@@ -307,7 +307,15 @@
 	name = "beer goggles"
 	icon_state = "sunhudbeer"
 	desc = "A pair of sunglasses outfitted with apparatus to scan reagents, as well as providing an innate understanding of liquid viscosity while in motion."
```
Comment (Fikou) [inline code/modules/clothing/glasses/_glasses.dm:309]: akhem

**Comment by MetalClone** [comment]
as quaint as it is for the bartender to have a skillchip. The beer goggles throw isn't on the same level of utility and power as the CQC the chef gets. And this removes the situation where people who benefit from the spare set of beer goggles being given to them, namely the chef, stand-in bartenders, and occasionally chemists. Also as quaint as a gimmick item is, I genuinely don't anyone would ever use them other than on a hostage which sounds funny at first, but is probably one-sided in terms of whose laughing.

**Comment by CMDR-Gungnir** [comment]
> as quaint as it is for the bartender to have a skillchip. The beer goggles throw isn't on the same level of utility and power as the CQC the chef gets. And this removes the situation where people who benefit from the spare set of beer goggles being given to them, namely the chef, stand-in bartenders, and occasionally chemists. Also as quaint as a gimmick item is, I genuinely don't anyone would ever use them other than on a hostage which sounds funny at first, but is probably one-sided in terms of whose laughing.

I play Bartender on a static that uses Nearsighted, this is based tho.

**Comment by JohnFulpWillard** [comment]
Should beer goggles even remain as a subtype of scanning goggles after this?

**Comment by Dorsisdwarf** [comment]
I don't see why beergoggles have to be removed... they were a nice source of getting reagent measurement for service roles if there wasnt a Round Assigned Barkeeper

**Comment by Coastfront** [comment]
but why do beer goggles need to be turned into a shitty gag item for bartender to be given innate beer goggle reagent scanning, this isnt even comparable to cqc, one just lets you see reagents and throw glasses without obliterating them, and the other lets you put someone to sleep instantly at melee range

File: code/modules/clothing/glasses/_glasses.dm (hunk 2 above)
Comment (timothymtorres) [reply code/modules/clothing/glasses/_glasses.dm]: I was meaning to get to this but I forgot.  Fixed with 4101513c2f08147570369c89352774779cc7e556 ty

File: code/modules/clothing/glasses/_glasses.dm (hunk 1 above)
Comment (timothymtorres) [reply code/modules/clothing/glasses/_glasses.dm]: I saw that with other glasses but `update_eye_blur` doesn't do anything if you don't have the DRUNK TRAIT.  Regardless I will make it consistent with the other glasses.  Applied with 4101513c2f08147570369c89352774779cc7e556

**Comment by Mothblocks** [comment]
Pretty sure me and @tralezab both want ITEMS to be what gives you your abilities and such. That is what incentivizes people to stay looking like their job, I'm not convinced why this is a good idea

**Comment by CMDR-Gungnir** [comment]
> Pretty sure me and @tralezab both want ITEMS to be what gives you your abilities and such. That is what incentivizes people to stay looking like their job, I'm not convinced why this is a good idea

As someone who runs Nearsighted, I like the idea personally.

**Comment by Mothblocks** [comment]
Yeah I'm going to close this unless I'm totally wrong about that, but we moved fast grab back to nitrile gloves instead of on a skill chip for smiilar reasons

**Pull request closed without merging**